buildSrc/src/main/kotlin/org/jetbrains/desktop/buildscripts/CollectWindowsArtifactsTask.kt

```diff
@@ -0,0 +1,32 @@
+package org.jetbrains.desktop.buildscripts
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.FileSystemOperations
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+import javax.inject.Inject
+
+abstract class CollectWindowsArtifactsTask @Inject constructor(
+    objectFactory: ObjectFactory,
+    private val fs: FileSystemOperations,
+) : DefaultTask() {
+    @get:InputFiles
+    val angleBinaries = objectFactory.fileCollection()
+
+    @get:InputFile
+    val nativeLibrary = objectFactory.fileProperty()
+
+    @get:OutputDirectory
+    val targetDirectory = nativeLibrary.map { it.asFile.parentFile }
```

Comment:
That looks a bit odd, depending on the provenance of `nativeLibrary` it could lead to two tasks sharing the output quite easily, which is a no-go in Gradle.
I would instead use a `targetDirectory` that has no relationship with `nativeLibrary`'s path, like something based on the `buildDirectory` instead, and copy both `angleBinaries` and `nativeLibrary` to it in the execution phase of the task.